deepchem/applications/synthesis.py

```diff
@@ -0,0 +1,185 @@
+from typing import Tuple, Dict, Any, Union, List
+
+_SettingsType = Dict[str, Union[int, float, str, bool]]
+
+
+class RetrosynthesisEngine:
+  """Retrosynthesis engine
+
+  This holds an interface to the AiZynthFinder retrosynthesis library
```

Comment:
As a couple of general review thoughts, it would be really useful if we could integrate `RetrosynthesisEngine` a little more tightly with DeepChem's API. @Suzukazole has been looking at exploring  molecular transformer models and ideally we could experiment with swapping in new models into `RetrosynthesisEngine`